### PR TITLE
coverage-select-card

### PR DESCRIPTION
**Pull request by chrisdugne**
## Detailed purpose of the PR

coverage select card

### Testing strategy

- [x] I ensured my code is tested: 
- [x] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

File: src/redux/actions/catalog/cards/select.test.js [hunk 1]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+      const options = {
+        services: {
+          Content: {
+            find: jest.fn()
+          }
+        }
+      };
+
+      // $FlowFixMe duplicated type between store and layer
+      options.services.Content.find.mockReturnValueOnce(Promise.resolve(level));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: why its not defined in options directly ?

Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: add an expect there please and use mockImplementationOnce so

File: src/redux/actions/catalog/cards/select.test.js [hunk 2]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+          Bundle: {
+            store: jest.fn(),
+            findById: jest.fn()
+          }
+        }
+      };
+
+      options.services.Content.find.mockReturnValueOnce(Promise.resolve());
+      options.services.Bundle.findById.mockReturnValueOnce(Promise.resolve());
+      options.services.Bundle.store.mockReturnValueOnce(Promise.resolve());
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same then above

File: src/redux/actions/catalog/cards/select.test.js [hunk 3]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+
+      // $FlowFixMe duplicated type between store and layer
+      options.services.Content.find.mockReturnValueOnce(Promise.resolve(level));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same than above

File: src/redux/actions/catalog/cards/select.test.js [hunk 4]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+    const options = {
+      services: {
+        Content: {
+          find: jest.fn()
+        }
+      }
+    };
+
+    // $FlowFixMe
+    options.services.Content.find.mockReturnValueOnce(Promise.resolve(null));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same then above

File: src/redux/actions/catalog/cards/select.test.js [hunk 5]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+        Content: {
+          find: jest.fn()
+        }
+      }
+    };
+
+    const badLevel = {
+      ...level,
+      universalRef: null
+    };
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: use createLevel from fixtures please

File: src/redux/actions/catalog/cards/select.test.js [hunk 6]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+    const modal = {
+      type: SHOW,
+      payload: {
+        errorType: ERROR_TYPE.NO_CONTENT_FOUND,
+        lastAction: expect.any(Function)
+      }
+    };
+
+    // $FlowFixMe
+    options.services.Content.find.mockReturnValueOnce(Promise.resolve(null));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same than above

File: src/redux/actions/catalog/cards/select.test.js [hunk 7]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+          Content: {
+            find: jest.fn()
+          }
+        }
+      };
+
+      const error = new Error('Chapter progression not created - no token or brand provided');
+
+      // $FlowFixMe
+      options.services.Content.find.mockReturnValueOnce(Promise.resolve(null));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same than above

File: src/redux/actions/catalog/cards/select.test.js [hunk 8]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+    };
+
+    const error = new NoContentFoundError(
+      `No Content Found for card with universalRef ${disciplineCard.universalRef}`
+    );
+
+    options.services.Content.find.mockReturnValueOnce(Promise.resolve());
+    options.services.Bundle.findById.mockReturnValueOnce(Promise.resolve());
+    options.services.Bundle.store.mockReturnValueOnce(Promise.resolve());
+    options.services.Content.find.mockReturnValueOnce(Promise.resolve());
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same than above

File: src/redux/actions/catalog/cards/select.test.js [hunk 9]
```diff
@@ -27,12 +24,340 @@ const chapterCard = createChapterCard({
...
+    const modal = {
+      type: SHOW,
+      payload: {
+        errorType: ERROR_TYPE.NO_CONTENT_FOUND,
+        lastAction: expect.any(Function)
+      }
+    };
+
+    // $FlowFixMe
+    options.services.Content.find.mockReturnValueOnce(Promise.resolve(null));
```
Comment (VincentCATILLON) [inline src/redux/actions/catalog/cards/select.test.js]: same than above (sorry for spamming)

File: src/redux/actions/catalog/cards/select.test.js (hunk 5 above)
Comment (chrisdugne) [reply src/redux/actions/catalog/cards/select.test.js]: already the case, we need to override `universalRef` here
```
const level = createCardLevel({ref: 'mod_1', status: CARD_STATUS.ACTIVE, label: 'Fake level'});
```

**Pull request merged**